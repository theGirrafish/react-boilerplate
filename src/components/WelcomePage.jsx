import React, {PureComponent} from 'react';
import Logo from '../assets/images/react.jpg';
import Background from '../assets/images/background.jpg';

class WelcomePage extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    const bgImageCSS = {
      height: '100%',
      backgroundImage: `url(${Background})`,
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover'
    }

    const titleCSS = {
      left: '0',
      top: '50%',
      width: '100%',
      color: '#000',
      textAlign: 'center',
      position: 'absolute'
    }

    const textCSS = {
      color: '#fff',
      padding: '18px',
      fontSize: '25px',
      letterSpacing: '10px'
    }

    return (
      <div>
        Gang
        <div style={bgImageCSS}>
          <div style={titleCSS}>
            <span style={textCSS}>Congratulations!</span>
            <span style={textCSS}>Looks like everything is up and running, now it's time to start coding</span>
          </div>
        </div>
      </div>
    );
  }
}

export default WelcomePage;
