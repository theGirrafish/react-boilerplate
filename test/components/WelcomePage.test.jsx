import React from 'react';
import renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';
import WelcomePage from '../../src/components/WelcomePage';
import Logo from '../assets/images/react.png';

describe('WelcomePage', () => {
  it('should match the existing snapshot', () => {
    const realDate = Date;
    const mockDate = new Date(1542531108880);
    Date = jest.fn(() => mockDate);

    const snapshot = renderer.create(<WelcomePage/>).toJSON();
    expect(snapshot).toMatchSnapshot();

    Date = realDate;
  });

  it('should match the time and date shown', () => {
    const wrapper = shallow(<WelcomePage/>);

    expect(wrapper.state().date.toLocaleTimeString()).toEqual((new Date()).toLocaleTimeString());
    expect(wrapper.state().date.toLocaleDateString()).toEqual((new Date()).toLocaleDateString());
  });

  it('should match the time and date shown', () => {
    var wrapper = mount(<WelcomePage/>);
    const time = wrapper.find('div.time').find('span.subtitle');

    time.props().onClick();
    expect(wrapper.state().displayTime).toEqual(false);
    expect(time.text()).toEqual((new Date()).toLocaleDateString());
  });
});
