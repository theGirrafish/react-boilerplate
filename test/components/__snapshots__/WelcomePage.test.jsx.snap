// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage should match the existing snapshot 1`] = `
<div
  className="background"
>
  <img
    className="react_logo"
    src="test-file-stub"
  />
  <div
    className="text"
  >
    <span
      className="title"
    >
      The React Boilerplate
    </span>
    <span
      className="subtitle"
    >
      Congratulations!
    </span>
    <span
      className="subtitle"
    >
      Looks like everything is up and running.
    </span>
  </div>
  <div
    className="time"
  >
    <span
      className="subtitle"
      onClick={[Function]}
    >
      08:51:48
    </span>
  </div>
</div>
`;
